src/instructlab/eval/ruler.py

```diff
@@ -0,0 +1,169 @@
+# Standard
+from typing import Any, Dict, List, Optional
+import json
+import os
+import pathlib
+
+# Third Party
+from lm_eval.evaluator import simple_evaluate
+
+# First Party
+from instructlab.eval.evaluator import Evaluator
+
+RULER_TASKS = [
+    "niah_single_1",
+    "niah_single_2",
+    "niah_single_3",
+    "niah_multikey_1",
+    "niah_multikey_2",
+    "niah_multikey_3",
+    "niah_multiquery",
+    "niah_multivalue",
+    "ruler_vt",
+    "ruler_cwe",
+    "ruler_fwe",
+    "ruler_qa_hotpot",
+    "ruler_qa_squad",
+]
+
+DEFAULT_MAX_LENGTH = 4096
+
+
+class RulerEvaluator(Evaluator):
+    """
+    Class definition for running RULER benchmarking tasks.
+    """
+
+    name = "ruler"
+
+    def __init__(
+        self,
+        model_path: Optional[str] = None,
+        output_file: Optional[str] = None,
+        tasks: list[str] = RULER_TASKS,
```

Comment:
In Python, you want to avoid using complex objects like Lists, Dicts, Sets, etc. as defaults when declaring functions. The reason is that you are signing the default as a pointer to the instantiated object (in this case `RULER_TASKS`). So the workaround we do is to have a function setup like:

```py
def foo(my_list: list = None):
  if not my_list:
    my_list = []
```

In the case of this `__init__` method, since `RULER_TASKS` is a list which already contains data, we would want to create a copy of it for this default, which we can do with the slice notation (`[:]`):

```py
def __init__(self, tasks: list[str] = None):
  if not tasks:
    # create copies of the strings in RULER_TASKS
    self.tasks = RULER_TASKS[:]
  else:
    self.tasks = tasks

```